packages/core/src/toolkit-provider/ToolkitProvider.tsx

```diff
@@ -137,16 +146,46 @@ export function ToolkitProvider({
     (Array.isArray(inheritedThemes) && inheritedThemes.length === 0);
   const density = densityProp ?? inheritedDensity ?? DEFAULT_DENSITY;
   const themeName = getThemeName(themesProp, inheritedThemes);
-  const themes: Theme[] = getTheme(themeName);
+  const themes: Theme[] = useMemo(() => getTheme(themeName), [themeName]);
```

Comment:
This definitely better.

@joshwooding Do we care that consumers may pass in an inline array of themes which could potentially be a new value each render? e.g. `<ToolkitProvider theme={['light', 'other']} >`

Reply:
I guess, we should add a comment to say theme should be stable

Reply:
As we said before the alternative is to stringily it to use as a dependency, but that could be overkill